### ISSUE TITLE
Backend: Support timm catalog integration and dynamic manifest resolution

### ISSUE THREAD

**Issue by itallix** [open-edge-platform/training_extensions#7100]
Integrate the offline `timm` snapshot catalog from #7095 into the backend service to enable dynamic model selection:
- **Manifest Resolution:** Implement `TimmManifestProvider` to lazily construct `ModelManifest` instances on demand from `timm_catalog_snapshot.json` while serving a single synthetic timm card entry in the standard classification architecture listing.
- **Catalog Search API:** Expose a paginated `GET /api/model_architectures/timm/backbones` endpoint supporting search, limit/offset pagination, and family facet filtering.
- **Schema Updates:** Extend backend domain models (`ModelManifest`) and DTOs to support `family`, `version`, and optional `timm_managed` weight sources.
- **Config Conversion:** Update `GetiConfigConverter` to route `image-classification-timm-*` IDs to `timm_generic.yaml` and inject arch-specific data_input_params (`input_size`, `mean`, `std`).
- **Hyperparameter Handling:** Adapt `HyperparametersUpdater` to write the learning rate directly to model initialization arguments when no optimizer block is present in the recipe.

Refer to [design doc#section 4](https://github.com/open-edge-platform/geti/blob/develop/library/docs/design/timm-integration.md#4-required-backend-changes) for more details

**Comment by leoll2** [comment]
I suggest to also add an endpoint `GET /api/model_architectures/timm/families` that returns the list of model families, to let the UI know which options to display in the family filter. The alternative would be to hardcode the family values in the UI, but then we should revise this list everytime we upgrade timm.

**Comment by itallix** [comment]
Since we decided to use 3 cascading dropdowns (`family`, `variant`, & `tag`) instead of a single searchable field, adding search/pagination endpoint isn't necessary. Instead, I'm considering the following two options. 

@jpggvilaca let me know which approach you prefer:

1. **Single full catalog fetch** (fetch the entire catalog hierarchy once and let the client handle cascading dropdown logic locally):
- `GET /api/model_architectures/timm/catalog`:
```json
{
  "beit": {
    "base_patch16_224": [
      "in22k_ft_in22k",
      "in22k_ft_in22k_in1k"
    ],
    "base_patch16_384": [
      "in22k_ft_in22k_in1k"
    ],
    "beitv2_base_patch16_224": [
      "in1k_ft_in1k",
      "in1k_ft_in22k",
      "in1k_ft_in22k_in1k"
    ]
}
```
- `GET /api/model_architectures/timm/manifest?family={family}&variant={variant}&pretrained-tag={pretrained-tag}` -> ModelArchitectureView

2. **Individual endpoint per dropdown** (fetch options dynamically step-by-step as the user interacts with each dropdown):
- `GET /api/model_architectures/timm/families` -> list[str]
- `GET /api/model_architectures/timm/families/{family}/variants` -> list[str]
- `GET /api/model_architectures/timm/families/{family}/variants/{variant}/pretrained-tags` - list[str]
- `GET /api/model_architectures/timm/manifest?family={family}&variant={variant}&pretrained-tag={pretrained-tag}` -> ModelArchitectureView

**Comment by jpggvilaca** [comment]
How big is the entire response? And how many db queries would be executed in each option? Im trying to see which one of the options if worse for the backend as well. For the client side i'd much rather have the whole thing in one go, and i'll parse it on the client side. This way the experience will be snappy instead of 3 loadings, specially because the dropdowns depend on each other. family-variant-tag. So, (1) for me

**Comment by leoll2** [comment]
@itallix The response model that you propose in approach (1) can't be represented and validated against a schema, because the keys (`beit`, `base_patch16_224`, ...) are not statically defined. An equivalent representation that can be validated would be:
```json
{
"families": [
  {
    "name": "beit",
    "variants": [ ... ],
  },
]
}
```
Still, I don't think it is the best option due to the very large response size (1400+ objects). Option (2) looks more reasonable and scalable to me.

**Comment by itallix** [comment]
The whole json tree is 35KB now. There is no DB involved, we use JSON file as the source. From the backend perspective, Option 1 gives us less boilerplate. It's indeed 1400 objects, but it's structured into a hierarchy where each level [1; ~20] entries.
The downside of Option 2 is unnecessary network calls.

**Comment by jpggvilaca** [comment]
I am also ok with option 2, simpler on our side

**Comment by itallix** [comment]
Ok, let's proceed with Option 2.
